File: src/main/kotlin/randy/vintrace/endpoints/models/BreakdownType.kt
package randy.vintrace.endpoints.models

import com.fasterxml.jackson.annotation.JsonValue

enum class BreakdownType(@JsonValue val key: String) {
    YEAR("year"),
    VARIETY("variety"),
    REGION("region"),
    YEAR_VARIETY("year-variety"),
    REGION_VARIETY("region-variety"),
}

File: frontend/src/clients/types/BreakdownType.ts
export const BreakdownType = {
    YEAR: { key: "year", humanReadable: "Year" },
    VARIETY: { key: "variety", humanReadable: "Variety" },
    REGION: { key: "region", humanReadable: "Region" },
    YEAR_VARIETY: { key: "year-variety", humanReadable: "Year & Variety" },
    REGION_VARIETY: { key: "region-variety", humanReadable: "Region & Variety" },
} as const;

// Bit of a hack to create an enum-like type with complex values.
export type BreakdownType = typeof BreakdownType[keyof typeof BreakdownType]
